R/redist.R

```diff
@@ -807,25 +1466,38 @@ redist.mcmc <- function(adjobj, popvec, nsims, algorithm = c("mcmc", "anneal"),
                         constraint = NULL, constraintweights = NULL,
                         compactness_metric = "fryer-holden",
                         betaseq = "powerlaw", betaseqlength = 10,
-                        betaweights = NULL, 
+                        betaweights = NULL, beta = -10, freq = 100,
                         adjswaps = TRUE, rngseed = NULL, maxiterrsg = 5000,
                         adapt_lambda = FALSE, adapt_eprob = FALSE,
                         contiguitymap = "rooks", exact_mh = FALSE, savename = NULL,
                         verbose = TRUE,
                         num_hot_steps = 40000, num_annealing_steps = 60000,
                         num_cold_steps = 20000, ncores = 1
 ){
-  
+
   if(verbose){
     ## Initialize ##
     divider <- c(paste(rep("=", 20), sep = "", collapse = ""), "\n")
-    
+
     cat("\n", append = TRUE)
     cat(divider, append = TRUE)
     cat("redist.mcmc(): Automated Redistricting Simulation Using
          Markov Chain Monte Carlo\n\n", append = TRUE)
   }
-  
+
+  if(algorithm == "mpi"){
+  ## Check if Rmpi library is installed
+  if (!requireNamespace("Rmpi", quietly = TRUE)) {
+      stop("You must install package 'Rmpi' to use this function. Please install it if you wish to continue."

```

Comment:
Very minor point, but you might want to change the language of this to be "to use the MPI argument" rather than "to use this function", since it's no longer a standalone function.